File: Backend/src/main/java/edu/icet/dto/Appointment.java
package edu.icet.dto;

import edu.icet.enums.Status;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Appointment {
    private int appointmentId;
    private String appointmentDate;
    private String appointmentTime;
    private String location;
    private String reasonVisit;
    private Status status;
    private String dateTimeStats;
    private int patientId;
    private int doctorId;
}

File: Frontend/src/app/model/appointment.ts
export class appointment {
    appointmentId: number;
    appointmentDate: string;
    appointmentTime: string;
    location: string;
    reasonVisit: string;
    status: string;
    dateTimeStats: string;
    patientId: number;
    doctorId: number;
    patientName: string;
    doctorName: string;

    constructor(
        appointmentId: number,
        appointmentDate: string,
        appointmentTime: string,
        location: string,
        reasonVisit: string,
        status: string,
        dateTimeStats: string,
        patientId: number,
        doctorId: number,
        patientName: string,
        doctorName: string
    ) {
        this.appointmentId = appointmentId;
        this.appointmentDate = appointmentDate;
        this.appointmentTime = appointmentTime;
        this.location = location;
        this.reasonVisit = reasonVisit;
        this.status = status;
        this.dateTimeStats = dateTimeStats;
        this.patientId = patientId;
        this.doctorId = doctorId;
        this.patientName = patientName;
        this.doctorName = doctorName;
    }
}
